### PR TITLE
[Merged by Bors] - tortoise: add test case for out of order layer

### PR DESCRIPTION
**Pull request by dshulyak**
## Motivation

Tortoise can't handle an out-of-order layer without reloading votes from all layer past out-of-order layer.
If we want to support such use case we need to re-process blocks from all layers after the one that is out of order.

Closes #

## TODO

- [ ] Explain motivation or link existing issue(s)
- [ ] Test changes and document test plan
- [ ] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

File: tortoise/sim/generator.go [hunk 1]
```diff
@@ -84,6 +85,7 @@ type Generator struct {
 	State State
 
 	nextLayer   types.LayerID
+	reordered   map[types.LayerID]types.LayerID
```
Comment (countvonzero) [inline tortoise/sim/generator.go:89]: can you add comment of what key/value supposed to be? thanks

File: tortoise/verifying_tortoise_test.go [hunk 2]
```diff
@@ -3074,12 +3074,39 @@ func TestRerunMissingLayer(t *testing.T) {
...
+	cfg := defaultConfig(t, s.State.MeshDB, s.State.AtxDB)
+	tortoise := NewVerifyingTortoise(ctx, cfg)
+
+	var (
+		inorder  types.LayerID
+		verified types.LayerID
+	)
+	for _, lid := range sim.GenLayers(s,
+		sim.WithSequence(1),
+		sim.WithSequence(1, sim.WithNextReorder(1)),
```
Comment (countvonzero) [inline tortoise/verifying_tortoise_test.go:3065]: can you add comments about what `WithNextReorder()` do or give a concrete example for the layers generated by these sequences? i find this hard to understand quickly.

Comment (dshulyak) [reply tortoise/verifying_tortoise_test.go:3065]: in this case the layers are 8, 10, 9, 11, 12 . i will try to clarify that in the WithNextReorder, it basically defines the delay for inorder layer

**Review by countvonzero — APPROVED** [review]

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**